cluster-autoscaler/main.go

```diff
@@ -252,6 +263,23 @@ func createAutoscalingOptions() config.AutoscalingOptions {
 	if *maxDrainParallelismFlag > 1 && !*parallelDrain {
 		klog.Fatalf("Invalid configuration, could not use --max-drain-parallelism > 1 if --parallel-drain is false")
 	}
+
+	// in order to avoid inconsistent deletion thresholds for the legacy planner and the new actuator, the max-empty-bulk-delete,
+	// and max-scale-down-parallelism flags must be set to the same value.
+	if isFlagPassed("max-empty-bulk-delete") && !isFlagPassed("max-scale-down-parallelism") {
+		// Set maxScaleDownParallelismFlag = maxEmptyBulkDeleteFlag to avoid inconsistent deletion thresholds for the legacy planner and the new actuator.
+		*maxScaleDownParallelismFlag = *maxEmptyBulkDeleteFlag
+	} else if !isFlagPassed("max-empty-bulk-delete") && isFlagPassed("max-scale-down-parallelism") {
+		// Set maxEmptyBulkDeleteFlag = maxScaleDownParallelismFlag to avoid inconsistent deletion thresholds for the legacy planner and the new actuator.
+		*maxEmptyBulkDeleteFlag = *maxScaleDownParallelismFlag
+	} else {
+		maxDeletionSize := int(math.Max(float64(*maxEmptyBulkDeleteFlag), float64(*maxScaleDownParallelismFlag)))
+		*maxEmptyBulkDeleteFlag = maxDeletionSize
+		*maxScaleDownParallelismFlag = maxDeletionSize
+
+		klog.Warning("setting maxEmptyBulkDeleteFlag and maxScaleDownParallelismFlag to the max of the two flags, in order to avoid inconsistent deletion thresholds for the legacy planner and the new actuator.")
```

Comment:
Couple of nits:
- I'd use flag names, rather than variable names in the log.
- Maybe log the values of flags and the resulting value being used? That should be more readable.
- Let's start with capital letter.

Reply:
Addressed, but rather than logging here, i am logging in the max-empty-bulk-delete block